tests/integration/test_wazuh_db/test_wazuh_db.py

```diff
@@ -64,79 +72,156 @@ def clean_registered_agents():
+    """Check that every input agent message in wazuh-db socket generates the adequate output to wazuh-db socket.
 
-    Parameters
-    ----------
-    test_case : list
-        List of test_case stages (dicts with input, output and stage keys).
+    Args:
+        test_case(list): List of test_case stages (dicts with input, output and stage keys).
     """
     for index, stage in enumerate(test_case):
-        if 'ignore' in stage and stage['ignore'] == "yes":
+        if 'ignore' in stage and stage['ignore'] == 'yes':
             continue
 
-        expected = stage['output']
-        receiver_sockets[0].send(stage['input'], size=True)
-        response = receiver_sockets[0].receive(size=True).decode()
+        command = stage['input']
+        expected_output = stage['output']
+
+        response = execute_wazuh_db_query(command)
 
         if 'use_regex' in stage and stage['use_regex'] == 'yes':
-            match = True if regex_match(expected, response) else False
+            match = True if regex_match(expected_output, response) else False
         else:
-            match = (expected == response)
+            match = (expected_output == response)
         assert match, 'Failed test case stage {}: {}. Expected: {}. Response: {}' \
-            .format(index + 1, stage['stage'], expected, response)
+            .format(index + 1, stage['stage'], expected_output, response)
 
 
-def test_wazuh_db_create_agent(restart_wazuh, clean_registered_agents, configure_sockets_environment, connect_to_sockets_module):
-    """Check that Wazuh DB creates the agent database when a query with a new agent ID is sent"""
-    test = {"name": "Create agent",
-            "description": "Wazuh DB creates automatically the agent's database the first time a query with a new agent"
-                           " ID reaches it. Once the database is created, the query is processed as expected.",
-            "test_case": [{"input": "agent 999 syscheck integrity_check_left",
-                           "output": "err Invalid FIM query syntax, near 'integrity_check_left'",
-                           "stage": "Syscheck - Agent does not exits yet"}]}
-    test_wazuh_db_messages(clean_registered_agents, restart_wazuh, configure_sockets_environment, connect_to_sockets_module, test['test_case'])
-    assert os.path.exists(os.path.join(WAZUH_PATH, 'queue', 'db', "999.db"))
+@pytest.mark.parametrize('test_case',
+                         [case['test_case'] for module_data in global_module_tests for case in module_data[0]],
+                         ids=[f"{module_name}: {case['name']}"
+                              for module_data, module_name in global_module_tests
+                              for case in module_data]
+                         )
+def test_wazuh_db_messages_global(configure_sockets_environment, connect_to_sockets_module, 
+                                  stop_all_daemons, test_case):
```

Comment:
It is not necessary to use `configure_sockets_environment`
```suggestion
def test_wazuh_db_messages_global(restart_wazuh, connect_to_sockets_module, test_case):
```